internal/operands/vm-console-proxy/reconcile.go

```diff
@@ -271,10 +331,43 @@ func getVmConsoleProxyImage() string {
 	return common.EnvOrDefault("VM_CONSOLE_PROXY_IMAGE", "quay.io/kubevirt/vm-console-proxy:v0.1.0")
 }
 
+func findResourcesUsingLabels[PtrL interface {
+	*L
+	client.ObjectList
+}, PtrT interface {
+	*T
+	client.Object
+}, L any, T any](ctx context.Context, name string, cli client.Client, itemsFunc func(list PtrL) []T) ([]client.Object, error) {
```

Comment:
Would love to see this simplified with better variable names if possible but that's something for a follow up as this works as expected at the moment.